emcc.py

```diff
@@ -2207,6 +2212,17 @@ def phase_linker_setup(options, state, newargs):
     if options.use_closure_compiler is None and settings.TRANSPILE_TO_ES5:
       diagnostics.warning('transpile', 'enabling transpilation via closure due to browser version settings.  This warning can be suppressed by passing `--closure=1` or `--closure=0` to opt into our explicitly.')
 
+  # https://caniuse.com/class: EDGE:13 FF:45 CHROME:49 SAFARI:9
+  supports_es6_classes = (settings.MIN_EDGE_VERSION >= 13 and
+                          settings.MIN_FIREFOX_VERSION >= 45 and
+                          settings.MIN_CHROME_VERSION >= 49 and
+                          settings.MIN_SAFARI_VERSION >= 90000 and
+                          settings.MIN_IE_VERSION == 0x7FFFFFFF)
+
+  if settings.EXCEPTION_STACK_TRACES and not supports_es6_classes:
+    diagnostics.warning('transpile', '-sEXCEPTION_STACK_TRACES requires an engine that support ES6 classes.')
+    settings.EXCEPTION_STACK_TRACES = 0
```

Comment:
Indeed, this only applies to Emscripten EH. Since `ASSERTIONS` implies `EXPORT_EXCEPTION_HANDLING_HELPERS` we have to make sure that targeting old engines still disables this option. Otherwise, it could still cause issues with simple programs, for example:
```console
$ echo -e 'int main(){return 0;}' | ./emcc -xc -sMIN_CHROME_VERSION=48 -o test.js -
emcc: warning: enabling transpilation via closure due to browser version settings.  This warning can be suppressed by passing `--closure=1` or `--closure=0` to opt into our explicitly. [-Wtranspile]
$ node test.js
/home/kleisauke/emscripten/test.js:75
      throw ex;
      ^

ReferenceError: $jscomp is not defined
```
(due to https://github.com/google/closure-compiler/issues/1138)

AFAIK, such construction would not be necessary with Wasm EH, since that is currently opt-in and not enabled by default (unlike `ASSERTIONS`, which is enabled by default).

Reply:
So shouldn't this warning only be generated when emscripten EH is used?  Right now it seems like I could see with warning with wasm-eh enabled, which is not correct, right?

Reply:
Commit 3b72a3592d6c52f45f0405e45dc90cb2417f0d5e avoids the emit of this transpile warning with `-fwasm-exceptions`.